bin/tests/cli.rs

```diff
@@ -0,0 +1,82 @@
+use assert_cmd::prelude::*;
+use predicates::prelude::*;
+
+use tempfile::tempdir;
+
+use std::fs::File;
+use std::io::Write;
+use std::process::Command;
+use std::thread;
+
+#[test]
+fn api_key_missing() {
+    let mut cmd = Command::cargo_bin("logdna-agent").unwrap();
```

Comment:
We seem to instantiate this object in many places. Does it make sense to create it once and have it available for the entire program?

Reply:
It's the initial state of a mutable builder object which we configure differently in each test, see the next line vs it's equivalent in the api_key_present test. Each of the methods takes a &mut, so to reuse it we'd need to clone it each time before we called any methods and unfortunately it doesn't impl Clone.

We _could_ factor it out into a function to reduce the boilerplate a little?

Reply:
I should have payed closer attention. I'm assuming the builder object is modified in the steps below in which case my suggestion doesn't make sense. Wrapping a function around it won't save much of any boilerplate... so it's probably fine to leave as is